packages/core-commands/src/site-editor-navigation-commands.js

```diff
@@ -56,8 +57,21 @@ const getNavigationCommandLoaderPerPostType = ( postType ) =>
 			[ supportsSearch, search ]
 		);
 
+		/*
+		 * wp_template and wp_template_part endpoints do not support per_page or orderby parameters.
+		 * We need to sort the results based on the search query to avoid removing relevant
+		 * records below using .slice().
+		 */
+		const orderedRecords = useMemo(
+			() =>
+				supportsSearch
+					? records
+					: orderEntityRecordsBySearch( records || [], search ),
+			[ supportsSearch, records, search ]
+		);
```

Comment:
What do you think about moving slicing logic here as well? This way, "order" and "limit" will be located in the same place.

Reply:
Something like:

```js
const orderedRecords = useMemo( () => {
	if ( supportsSearch ) {
		return records;
	}

	return orderEntityRecordsBySearch( records, search ).slice( 0, 10 );
}, [ supportsSearch, records, search ] );
```

The `orderEntityRecordsBySearch` will always return an `array`, and the `records` argument can be `undefined`.

Reply:
Makes sense! I made an additional change to the above to ensure that `orderedRecords` is always an `array` so no check is required on the `.map()` https://github.com/WordPress/gutenberg/pull/53286/commits/965949851526f59bafbbf3f49dde23b5b17fb27a